pandera/checks.py

```diff
@@ -438,22 +463,49 @@ def __repr__(self):
         )
 
 
+_T = TypeVar("_T", bound=_CheckBase)
+
+
 class _CheckMeta(type):  # pragma: no cover
     """Check metaclass."""
 
+    # FIXME: this should probably just be moved to _CheckBase
+
+    REGISTERED_CUSTOM_CHECKS: Dict[str, Callable] = {}  # noqa
```

Comment:
This comment got moved by black. Essentially, I think we could merge the `_CheckMeta` with `_CheckBase`. The metaclass actually confuses `mypy`, and I don't see the benefit of using this mixin pattern here.

Reply:
If you move `__getattr__` to `_CheckBase` it's going to act on instances of the class and not on the class itself.

Reply:
Right. Metaclasses confuse me, had to look up the descriptor protocol docs again. 👍

Reply:
Ok, I'm proposing we just move the metaclass to `_CheckBase`. It's not usable on anything else anyway because it requires a name property for example. I'll fix it and remove the fixme comment.